src/dmd/utils.d

```diff
@@ -138,3 +139,44 @@ const(char)[] toDString (const(char)* s) pure nothrow @nogc
 {
     return s ? s[0 .. strlen(s)] : null;
 }
+
+/**
+   Copy the content of `src` into a C-string ('\0' terminated) then call `dg`
+
+   The intent of this function is to provide an allocation-less
+   way to call a C function using a D slice.
+   The function internally allocates a buffer if needed, but frees it on exit.
+
+   Note:
+     The argument to `dg` is `scope`. To keep the data around after `dg` exits,
+     one has to copy it.
+
+   Params:
+     src = Slice to use to call the C function
+     dg  = Delegate to call afterwards
+
+   Returns:
+     The return value of `T`
```

Comment:
Nit: For new Phobos code, we recommend to not use any white space or stars as indentation.

Reply:
That's absolutely hideous, but if it's the D style...

Reply:
I agree, it's hideous.